src/pipeformer/internal/templates/codebuild.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Logic for building the CodeBuild stack templates."""
+import string
+
+from troposphere import AWS_STACK_NAME, Output, Parameter, Ref, Sub, Tags, Template, codebuild, iam, s3
+
+from pipeformer.internal.resolve import InputResolver
+from pipeformer.internal.structures import Config
+from pipeformer.internal.util import reference_name, resource_name
+
+from . import project_tags
+
+
+def project_name(action_number: int) -> str:
+    """Construct the project logical resource name.
+
+    :param action_number: Unique count identifier for project in stack
+    :return: Logical resource name
+    """
+    return resource_name(codebuild.Project, string.ascii_letters[action_number])
+
+
+def _build_project(name: str, action: InputResolver, role: Ref, bucket: Ref, tags: Tags) -> codebuild.Project:
+    """Construct a CodeBuild project for the specified action.
+
+    :param name: Logical resource name to use for project
+    :param action: Action wrapped in an InputResolver
+    :param role: Reference to CodeBuild role
+    :param bucket: Reference to application resources bucket
+    :param tags: Tags to add to project
+    :return: Constructed project
+    """
+    return codebuild.Project(
+        name,
+        Name=Sub(f"${{{AWS_STACK_NAME}}}-{name}"),
+        ServiceRole=role,
+        Artifacts=codebuild.Artifacts(Type="CODEPIPELINE"),
+        Source=codebuild.Source(Type="CODEPIPELINE", BuildSpec=action.buildspec),
+        Environment=codebuild.Environment(
+            ComputeType=action.compute_type,
+            Type=action.environment_type,
+            Image=action.image,
+            EnvironmentVariables=[codebuild.EnvironmentVariable(Name="PIPEFORMER_S3_BUCKET", Value=bucket)]
+            + [codebuild.EnvironmentVariable(Name=key, Value=value) for key, value in action.env.items()],
+        ),
+        Tags=tags,
+    )
+
+
+def build(project: Config, stage: InputResolver) -> Template:
+    """Build a stack template for all CodeBuild actions in a CodePipeline stage.
+
+    :param project: PipeFormer project to build for
+    :param stage: Stage for which to construct CodeBuild projects
+    :return: Constructed template
+    """
+    resources = Template(
+        Description=f"CodeBuild projects for {stage.name} stage in pipeformer-managed project: {project.name}"
```

Comment:
Should `pipeformer` here be a reference to the prefix defined elsewhere, or do you want to keep it as a string literal?

Reply:
This is meant simply to identify the project as being managed by pipeformer, not to identify the stack prefix. So I think keeping it as a string literal is fine, yeah.